clrtrust.in

```diff
@@ -135,15 +159,7 @@ find_certs() {
     local dir=$1
     if [ -z $dir ]; then return 255; fi
     if [ ! -d $dir ]; then return 255; fi
-    find $dir -maxdepth 1 -type f | while read f; do
-        finger=$(openssl x509 -in "${f}" -noout -fingerprint -sha1 2>/dev/null)
-        if [ $? -ne 0 ]; then
-            1>&2 echo "WARNING: file ${f} is not a certificate"
-            continue
-        fi
-        finger=${finger#SHA1 Fingerprint=}
-        printf "%s\t%s\n" "${f}" "${finger}"
-    done
+    find $dir -maxdepth 1 -type f | "${CLR_HELPER_CMD}" -f
```

Comment:
Is there any sanity checking somewhere that filenames do not include embedded newlines and tabs?

If not, then can I suggest using

    find "$dir" -maxdepth 1 -type f -exec "${CLR_HELPER_CMD}" -f {} +

and making the helper program loop over its arguments? Another popular approach is to add a -z flag to the helper to accept NUL terminated filenames rather than newline terminated ones.

Reply:
Since there are few other places already which would fail anyways, I documented this as an issue #16. I will address it at some point, but not as part of this change.